app/pages/lab/workflow.cjsx

```diff
@@ -53,6 +54,11 @@ EditWorkflowPage = React.createClass
     deletionError: null
     workflowCreationInProgress: false
 
+  componentWillReceiveProps: (nextProps) ->
+    if nextProps.workflow isnt @props.workflow
+      ReactDOM.findDOMNode(@refs.defaultWorkflow).defaultChecked = nextProps.project.configuration.default_workflow is nextProps.workflow.id
```

Comment:
I don't think `componentWillReceiveProps` is necessary here, since `checked` can all be set in `render` for both of these inputs.